### ISSUE TITLE
Bug report: ansible.cfg::remote_port has no effect in 2.0

### ISSUE THREAD

**Issue by skyler** [ansible/ansible#11451]
##### Issue Type:

Bug Report
##### Ansible Version:

```
ansible 2.0.0 (devel 47be5b4166) last updated 2015/06/02 11:58:32 (GMT -400)
  lib/ansible/modules/core: (detached HEAD 600fc15b42) last updated 2015/02/05 14:09:13 (GMT -400)
  lib/ansible/modules/extras: (detached HEAD 77485f6c6a) last updated 2015/02/05 14:09:19 (GMT -400)
  v1/ansible/modules/core:  not found - use git submodule update --init v1/ansible/modules/core
  v1/ansible/modules/extras:  not found - use git submodule update --init v1/ansible/modules/extras
  configured module search path = None
```
##### Ansible Configuration:

No change.
##### Environment:

CentOS release 6.5 (Final)
##### Summary:

In 1.9 stable, `remote_port` in `ansible.cfg` specifies an alternate SSH port. When I switch to the `devel` branch, changing nothing else about my configuration, `remote_port` has no effect.

Here is my `ansible.cfg`

```
[defaults]
remote_port = 2200
log_path = /tmp/ansible.log
ask_sudo_pass = true
```
##### Steps To Reproduce:

(See summary)
##### Expected Results:

(See summary)
##### Actual Results:

(See summary)

**Comment by bennojoy** [comment]
@skyler  i am not able to reproduce this, where is the ansible.cfg placed ? and did you checkout the latest version from devel ? an ouput  ansible --version should help

**Comment by jimi-c** [comment]
Hi, this situation is no longer present in the devel branch (which will be ansible 2.0). Here is the output using a simple reproducer:

<pre>
# ANSIBLE_REMOTE_PORT=222 ansible -m ping -vvv awxlocal 
Using /etc/ansible/ansible.cfg as config file
<192.168.122.100> ESTABLISH SSH CONNECTION FOR USER: root
<192.168.122.100> EXEC ssh -C -q -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/tmp/%h-%r" -o Port=222 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 192.168.122.100 LANG=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 LC_CTYPE=en_US.UTF-8 /usr/bin/python
EXEC previous known host file not found for 192.168.122.100
EXEC previous known host file not found for 192.168.122.100
awxlocal | FAILED! => {
    "failed": true, 
    "msg": "ERROR! SSH Error: data could not be sent to the remote host. Make sure this host can be reached over ssh"
}
</pre>

As you can see above in the verbose output, the Port is set to 222, which is as-specified via the environment variable. The mechanism that picks that up is the exact same as the ansible.cfg, so there should no longer be any problem with setting it there (I tested, just to be sure).

Per our normal policy, we close bug report issues when we see the issue has been corrected in the devel branch, so we will close this issue now.

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!

**Comment by skyler** [comment]
I understand that you're unlikely to see comments on closed issues, but I'm reposting here just in case you do see it. Failing that, I'll post a message to the mailing list.

This is happening for me with the latest devel branch.

```
ansible --version
ansible 2.0.0 (devel 9aa4214417) last updated 2015/07/31 09:45:58 (GMT -400)
  lib/ansible/modules/core: (detached HEAD 30377b7103) last updated 2015/07/13 15:17:50 (GMT -400)
  lib/ansible/modules/extras: (detached HEAD f291e9fcc3) last updated 2015/07/13 15:17:50 (GMT -400)
  v1/ansible/modules/core: (detached HEAD f8d8af17cd) last updated 2015/07/13 14:45:04 (GMT -400)
  v1/ansible/modules/extras: (detached HEAD 495ad450e5) last updated 2015/07/13 14:45:08 (GMT -400)
  config file = /home/vagrant/ansible/local/ansible.cfg
  configured module search path = None
```

`ansible.cfg`:

```
[defaults]
remote_port = 2200
log_path = /tmp/ansible.log
ask_sudo_pass = true
```

My inventories/local file:

```
[webservers]
#localhost                 ansible_connection=local
192.168.57.2 ansible_ssh_user=skyler
```

Trying to ping:

```
ansible -m ping -i inventories/local webservers -vvv
Using /home/vagrant/ansible/local/ansible.cfg as config file
SUDO password:
<192.168.57.2> ESTABLISH CONNECTION FOR USER: skyler on PORT 22 TO 192.168.57.2
192.168.57.2 | FAILED! => {
    "failed": true,
    "msg": "ERROR! [Errno 111] Connection refused"
}
```

Switching back to `stable-1.9`

```
~/ansible/local (vault)> ansible --version
ansible 1.9.2 (stable-1.9 6c21f3c0fd) last updated 2015/07/31 09:45:36 (GMT -400)
  lib/ansible/modules/core: (detached HEAD 30377b7103) last updated 2015/07/13 15:17:50 (GMT -400)
  lib/ansible/modules/extras: (detached HEAD f291e9fcc3) last updated 2015/07/13 15:17:50 (GMT -400)
  v2/ansible/modules/core: (detached HEAD 85c8a892c8) last updated 2015/07/13 15:17:51 (GMT -400)
  v2/ansible/modules/extras: (detached HEAD 70ea058563) last updated 2015/07/13 15:17:51 (GMT -400)
  configured module search path = None
~/ansible/local (vault)> ansible -m ping -i inventories/local webservers -vvv
SUDO password:
<192.168.57.2> ESTABLISH CONNECTION FOR USER: skyler on PORT 2200 TO 192.168.57.2
<192.168.57.2> REMOTE_MODULE ping
<192.168.57.2> EXEC /bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1438351080.23-188797857200230 && chmod a+rx $HOME/.ansible/tmp/ansible-tmp-1438351080.23-188797857200230 && echo $HOME/.ansible/tmp/ansible-tmp-1438351080.23-188797857200230'
<192.168.57.2> PUT /tmp/tmplN7GIu TO /home/skyler/.ansible/tmp/ansible-tmp-1438351080.23-188797857200230/ping
<192.168.57.2> EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_CTYPE=en_US.UTF-8 /usr/bin/python /home/skyler/.ansible/tmp/ansible-tmp-1438351080.23-188797857200230/ping; rm -rf /home/skyler/.ansible/tmp/ansible-tmp-1438351080.23-188797857200230/ >/dev/null 2>&1'
192.168.57.2 | success >> {
    "changed": false,
    "ping": "pong"
}
```

**Comment by amenonsen** [comment]
I can confirm that this behaviour exists with current devel, both with `ANSIBLE_REMOTE_PORT` and setting `remote_port` in the `[defaults]` section of `ansible.cfg`. The port doesn't seem to be set in the `PlayContext` where `ssh.py` is trying to read it from, but I haven't investigated further.

**Comment by bcoca** [comment]
fixed via f16b745f172a92e0357491bd5fdfea111de50089
